shared/js/background/allow-pages.es6.js

```diff
@@ -0,0 +1,96 @@
+const tdsStorage = require('./storage/tds.es6')
+const settings = require('./settings.es6')
+
+/**
+ * Creates an 'allow page' rule for given domain. Our extension will not try to upgrade it or block
+ * any trackers on that domain. Trackers from that domain loaded on other pages will still be blocked.
+ * @param {string} domain
+ */
+function addDomainToSafelist (domain) {
+    let res, rej
+    const promise = new Promise((resolve, reject) => { res = resolve; rej = reject })
+
+    chrome.declarativeNetRequest.addAllowedPages([`*://*.${domain}/*`], () => {
+        if (chrome.runtime.lastError) {
+            console.warn(`Error adding safelist entry.`, domain, chrome.runtime.lastError.message)
+            rej()
+            return
+        }
+
+        res()
+    })
+
+    return promise
+}
+
+/**
+ * Removes matching 'allow page' rule
+ * @param {string} domain
+ */
+function removeDomainFromSafelist (domain) {
+    let res, rej
+    const promise = new Promise((resolve, reject) => { res = resolve; rej = reject })
+
+    chrome.declarativeNetRequest.removeAllowedPages([`*://*.${domain}/*`], () => {
+        if (chrome.runtime.lastError) {
+            console.warn(`Error removing safelist entry.`, domain, chrome.runtime.lastError.message)
+            rej()
+            return
+        }
+
+        res()
+    })
+
+    return promise
+}
+
+/**
+ * We have to make sure that domains safelisted by the user, and domains safelisted by our temporary safelist (broken pages)
+ * are in sync with actual 'allow page' rules
+ */
+function syncSafelistEntries () {
+    let res, rej
+    const promise = new Promise((resolve, reject) => { res = resolve; rej = reject })
+
+    chrome.declarativeNetRequest.getAllowedPages(allowedPagesRules => {
```

Comment:
:red_circle: Has the API changed for this? On Chromium 82 I see:

![Screenshot from 2020-02-24 16-52-53](https://user-images.githubusercontent.com/1431577/75173014-20452e80-5726-11ea-8d89-24cf07c7b66e.png)